Sources/NIO/EventLoop.swift

```diff
@@ -566,12 +566,22 @@ extension EventLoop {
     /// Creates and returns a new `EventLoopFuture` that is already marked as success. Notifications will be done using this `EventLoop` as execution `NIOThread`.
     ///
     /// - parameters:
-    ///     - result: the value that is used by the `EventLoopFuture`.
+    ///     - value: the value that is used by the `EventLoopFuture`.
     /// - returns: a succeeded `EventLoopFuture`.
     @inlinable
     public func makeSucceededFuture<Success>(_ value: Success, file: StaticString = #file, line: UInt = #line) -> EventLoopFuture<Success> {
         return EventLoopFuture<Success>(eventLoop: self, value: value, file: file, line: line)
     }
+    
+    /// Creates and returns a new `EventLoopFuture` with state arbitrated from a provided `Result`. Notifications will be done using this `EventLoop` as execution `NIOThread`.
+    ///
+    /// - parameters:
+    ///     - value: the `Result` whose state will arbitrate the resultant `EventLoopFuture`.
+    /// - returns: a succeeded or failed `EventLoopFuture`.
+    @inlinable
+    public func makeResultantFuture<T>(_ value: Result<T, Error>, file: StaticString = #file, line: UInt = #line) -> EventLoopFuture<T> {
```

Comment:
I think I’d prefer `makeCompletedFuture`.

Reply:
I toyed with that actually - just slightly concerned “completed” sounds slightly closer to “succeeded” than being agnostic about what’s in the Result?

Reply:
I'm +1 on `makeCompletedFuture`; there's prior art in NIO around futures/promise accepting a `Result` type: `EventLoopFuture<Value>.whenComplete` and `EventLoopPromise<Value>.completeWith`, for example.